lib/bundler/shared_helpers.rb

```diff
@@ -136,6 +136,10 @@ def print_major_deprecations!
       major_deprecation("Bundler will only support rubygems >= 2.0, you are running #{Bundler.rubygems.version}")
     end
 
+    def check_home_dir_permissions
+      raise PathError, "There was an error while trying to use your home path #{Dir.home}" unless File.writable?(Dir.home)
```

Comment:
this should specifically mention that the home directory must exist & be writable